pkg/operator2/oauth.go

```diff
@@ -59,11 +63,21 @@ func (c *authOperator) handleOAuthConfig(
 
 	emptyTemplates := configv1.OAuthTemplates{}
 	if configTemplates := oauthConfig.Spec.Templates; configTemplates != emptyTemplates {
+		klog.Infof("DBG: using spec templates")
 		templates = &osinv1.OAuthTemplates{
 			Login:             syncData.addTemplateSecret(configTemplates.Login, loginField, configv1.LoginTemplateKey),
 			ProviderSelection: syncData.addTemplateSecret(configTemplates.ProviderSelection, providerSelectionField, configv1.ProviderSelectionTemplateKey),
 			Error:             syncData.addTemplateSecret(configTemplates.Error, errorField, configv1.ErrorsTemplateKey),
 		}
+	} else {
```

Comment:
move all of this logic out to its own file

and you need to handle each user configuration separately - you can set just one of the templates